### DISCUSSION TITLE
Register device/deviceprofile

### DISCUSSION THREAD

**Discussion by NguyenKhanh27** [edgexfoundry/community#96] [Q&A]
Hi all, I would like to ask about device/deviceprofile registeration. I have simulated a case that there is power loss after registering a device/deviceprofile. The behavior was strange, UI has loaded the device/profile that I POST into core-metadata, however, when I plug out and in power input, it will restart to run those in docker and the device/profile that I have registered were loss. In another case, if I register then use docker compose down, and later use docker compose up, the devices and profiles registered  are still there, even if I plug in and out power, it is still there. I believe it affect something with volume created by docker but Im unsure, I would appreciate any helps regarding to this strange behavior.
I'm using ver 2.3 with docker-compose, test with edgex-compose already

**Comment by cloudxxx8** [comment]
I believe it's about the setting of RedisDB.  It's the persisence layer of core-metadata, and it's a memory db.  If we shut it down gracefully, such as using docker compose down, it could always persist the data to the disk.  However, your scenario might cause data loss before it stores data to the disk.  Please see the article and try to configure RedisDB.
https://redis.io/docs/management/persistence/

**Reply by Safinko** [reply]
Please can you contact with me? I need some help in Edgex My mail is [safe.safe97@gmail.com]

**Reply by cloudxxx8** [reply]
Please consider to use the commercial version for the professional support
https://www.iotechsys.com/products/edge-xpert/
